kernel/mempool.c

```diff
@@ -113,6 +144,33 @@ void *k_calloc(size_t nmemb, size_t size)
 	return ret;
 }
 
+void *k_aligned_realloc(void *ptr, size_t align, size_t size)
+{
+	__ASSERT(align / sizeof(void *) >= 1 && (align % sizeof(void *)) == 0,
+		 "align must be a multiple of sizeof(void *)");
+
+	__ASSERT((align & (align - 1)) == 0, "align must be a power of 2");
```

Comment:
Not the fault of this PR, but I'm pretty sure these asserts are duplicating ones down in sys_heap.  Would be nice to avoid the needless code size bloat.  CONFIG_ASSERT needs rigor and attention to stay lightweight, otherwise people will stop using it.